File: frontend/src/services/itineraryService.ts
import axiosClient from './axiosClient';
import { Itinerary, ItineraryCreateRequest, ItineraryItem } from '../types/itinerary';
import { ApiResponse } from '../types/common';

export const itineraryService = {
  createItinerary: async (data: ItineraryCreateRequest): Promise<ApiResponse<Itinerary>> => {
    return await axiosClient.post('/itineraries', data);
  },

  getMyItineraries: async (): Promise<ApiResponse<Itinerary[]>> => {
    return await axiosClient.get('/itineraries/my-itineraries');
  },

  getItineraryById: async (id: number): Promise<ApiResponse<Itinerary>> => {
    return await axiosClient.get(`/itineraries/${id}`);
  },

  getItineraryByShareToken: async (token: string): Promise<ApiResponse<Itinerary>> => {
    return await axiosClient.get(`/itineraries/share/${token}`);
  },

  addItem: async (itineraryId: number, dayId: number, item: Partial<ItineraryItem>): Promise<ApiResponse<Itinerary>> => {
    return await axiosClient.post(`/itineraries/${itineraryId}/days/${dayId}/items`, item);
  },

  deleteItem: async (itineraryId: number, itemId: number): Promise<ApiResponse<Itinerary>> => {
    return await axiosClient.delete(`/itineraries/${itineraryId}/items/${itemId}`);
  },

  reorderItems: async (itineraryId: number, dayId: number, itemIdsInOrder: number[]): Promise<ApiResponse<Itinerary>> => {
    return await axiosClient.put(`/itineraries/${itineraryId}/days/${dayId}/reorder`, itemIdsInOrder);
  },

  generateShareLink: async (itineraryId: number): Promise<ApiResponse<{ shareToken: string; shareUrl: string }>> => {
    return await axiosClient.post(`/itineraries/${itineraryId}/generate-share-link`);
  },
};

File: backend/src/main/java/com/smarttravel/services/ItineraryService.java
package com.smarttravel.services;

import com.smarttravel.dto.request.ItineraryCreateRequest;
import com.smarttravel.dto.response.ItineraryItemDto;
import com.smarttravel.dto.response.ItineraryResponse;

import java.util.List;

public interface ItineraryService {
    ItineraryResponse createItinerary(ItineraryCreateRequest request, String userEmail);
    List<ItineraryResponse> getMyItineraries(String userEmail);
    ItineraryResponse getItineraryById(Long id, String userEmail);
    ItineraryResponse getItineraryByShareToken(String token);
    ItineraryResponse addItem(Long itineraryId, Long dayId, ItineraryItemDto itemDto, String userEmail);
    ItineraryResponse deleteItem(Long itineraryId, Long itemId, String userEmail);
    ItineraryResponse reorderItems(Long itineraryId, Long dayId, List<Long> itemIdsInOrder, String userEmail);
    String generateShareToken(Long itineraryId, String userEmail);
}
